src/libs/actions/Report.ts

```diff
@@ -1681,24 +1682,29 @@ function updateWriteCapabilityAndNavigate(report: Report, newValue: WriteCapabil
 
 /**
  * Navigates to the 1:1 report with Concierge
- *
- * @param ignoreConciergeReportID - Flag to ignore conciergeChatReportID during navigation. The default behavior is to not ignore.
  */
-function navigateToConciergeChat(ignoreConciergeReportID = false, shouldDismissModal = false) {
+function navigateToConciergeChat(shouldDismissModal = false, shouldPopCurrentScreen = false, checkIfCurrentPageActive = () => true) {
     // If conciergeChatReportID contains a concierge report ID, we navigate to the concierge chat using the stored report ID.
     // Otherwise, we would find the concierge chat and navigate to it.
-    // Now, when user performs sign-out and a sign-in again, conciergeChatReportID may contain a stale value.
-    // In order to prevent navigation to a stale value, we use ignoreConciergeReportID to forcefully find and navigate to concierge chat.
-    if (!conciergeChatReportID || ignoreConciergeReportID) {
```

Comment:
`ignoreConciergeReportID` is only used in https://github.com/Expensify/App/pull/36216/files#r1484170212. The reason behind this param is explained by the comment itself. To properly fix it, I clear the concierge ID when the user is logged out

Reply:
Which was the PR adding this code? Won't `conciergeChatReportID ` get removed when singed out along with other onyx data?

Reply:
It was added in https://github.com/Expensify/App/pull/29122.

> Won't conciergeChatReportID  get removed when singed out along with other onyx data?

No. `conciergeChatReportID` is a global(?) variable that is assigned when we receive the onyx data.
https://github.com/Expensify/App/blob/6914b1344ed1f8bd69037232a67e34ffea2581e4/src/libs/actions/Report.ts#L170

https://github.com/Expensify/App/blob/6914b1344ed1f8bd69037232a67e34ffea2581e4/src/libs/actions/Report.ts#L1092-L1098